path_config.go

```diff
@@ -189,6 +201,7 @@ func (b *kubeAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Requ
 		return nil, err
 	}
 
+	b.useAnnotationsAsAliasMetadata = data.Get("use_annotations_as_alias_metadata").(bool)
```

Comment:
wrt to the lifecycle comment above, if the plugin gets restarted, it will wipe the value of this setting, because the `kubeAuthBackend` will get recreated and we aren't durably storing the provided `use_annotations_as_alias_metadata` setting anywhere. If we move this into `kubeConfig` to include it in the storage write just above, then it will be durably stored and survive plugin restarts.